apis/opts/log_options.go

```diff
@@ -0,0 +1,36 @@
+package opts
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// ParseLogOptions parses [key=value] slice-type log options into map.
+func ParseLogOptions(driver string, logOpts []string) (map[string]string, error) {
+	opts, err := convertKVStringsToMap(logOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	if driver == "none" && len(opts) > 0 {
+		return nil, fmt.Errorf("don't allow to set logging opts for driver %s", driver)
+	}
+	return opts, nil
+}
+
+// convertKVStringsToMap converts ["key=value"] into {"key":"value"}
+//
+// TODO(fuwei): make it common in the opts.ParseXXX().
+func convertKVStringsToMap(values []string) (map[string]string, error) {
```

Comment:
How about encapsulating the function into pkg? 
Since I think this is a common function there.

Reply:
make senses.